### ISSUE TITLE
Blocks should be more descriptive

### ISSUE THREAD

**Issue by heikadog** [bluesky-social/social-app#3718]
**Is your feature request related to a problem? Please describe.**

Blocks should be more descriptive. I feel like there's not enough of an in-app explanation around blocks—things like blocking via a moderation list, how blocks affect a full conversation, and who blocks who.

**Describe the solution you'd like**

Simply put, there should be clearer messaging when these things happen:
- If a user blocks someone via a moderation list, it should say "Blocked by [List Name]" instead of just "Blocked post." 
- If a user blocks someone, users should still be able to see the full conversation if it, for instance, interrupts a thread, instead of isolating a reply. 
- If a user blocks or is blocked by someone, you should be able to see who the other user is outside of one spot in another person's replies that shows their display name. Maybe a "Blocked post" warning that still involves the regular data around a post, like the display name, username, and profile picture would be helpful here.

**Describe alternatives you've considered**

"Blocked post." on its own can be pretty effective if someone's trying to avoid a person, but it'd be useful to know more in-app context around blocks.

**Comment by Tahvohck** [comment]
Nuclear blocking (completely excising the user from your threads) is a feature, and it's especially useful for notable users that face repeated harassment so I'd be concerned about changing that functionality. Hiding the blocked person from just the user but not others is more of what the mute feature is for.

For the other two I fully support them. Extra information about the block context is good. However, I feel like for point 3 that extra information should be hidden behind a click-through. The mental strain of knowing a specific person you blocked is still harassing you can be pretty rough.

**Comment by scottdotweb** [comment]
See also these issues I opened: #1843 (allow viewing of posts by users you've muted or blocked) and #7017 (allow viewing profile information of users you've blocked).

**Comment by hexdotooo** [comment]
This is a duplicate of #2127.
